### DISCUSSION TITLE
提点意见

### DISCUSSION THREAD

**Discussion by lovesL** [tolking/element-pro-components#482] [Ideas]
1.CRUD 希望初始调用一次查询嘛。比如刚进页面默认内部调一次查询。建议加个参数
2.select的data希望能支持函数
![image](https://github.com/tolking/element-pro-components/assets/27960630/2825ec3b-9d05-40c4-9101-b57b29fec113)
类似这种吧 因为有时候数据可能来自后端 ，这就会涉及到同步异步的问题，改成支持函数的话就能避免了
3.通过ref.closeDialog()可以关闭Dialog但是不会触发defineCrudBeforeClose。目前发有个小的影响就是 先编辑弹框后然后关闭 然后在点击新增弹框会编辑框的数据 新增框也有了，但是这个可以通过defineCrudBeforeOpen 打开重置form.value 。影响不大 但是不知道会不会有其他影响。

**Comment by tolking** [comment]
1. 目前仅在page相关变化后触发load，初始就触发不符合逻辑
2. 增加函数无意义，直接传递一个响应式数据不就好了
3. 自己使用的时候触发不就好了

**Comment by lovesL** [comment]
嗯2 3点按你说的确实不是很重要，但是我觉得 如果不默认执行一次那刚进页面界面就是空白的感觉不是很好，当然也有解决的办法 我是这么写的 
![image](https://github.com/tolking/element-pro-components/assets/27960630/15a86626-d1c6-45a4-acb4-1d7c5ed125f5)
我想了一下其实如果改成初始化执行一次意义好像也不大，因为后续分页切换 ，点击查询和重置按钮 都需要重新请求一次，最后还是得定一个函数请求。我之前是写react的 我用的阿里的有一套组件库 我觉得他那个设计思路很好 ProComponent，他那个设计的类似CRUD 组件是把所有的请求都封装在了一个内部的函数里面。无论是分页 还是查询都只在那个函数里面触发。感觉很方便 也很简洁吧

**Reply by tolking** [reply]
请求与业务关联的太紧密了，你想的都是通过函数去请求列表。但也可以直接监听 页码或者搜索表单自动加载列表
